test/responses_unit_test.go

```diff
@@ -0,0 +1,222 @@
+			round, something, err = indexerC.LookupAssetByID(10).Do(context.Background())
+			response = models.LookupAssetByIDResponse{
+				CurrentRound: round,
+				Asset:        something.(models.Asset),
+			}
+		case "searchForAssets":
+			response, err = indexerC.SearchForAssets().Do(context.Background())
+		case "lookupAccountTransactions":
+			response, err = indexerC.LookupAccountTransactions("").Do(context.Background())
+		case "lookupAssetTransactions":
+			response, err = indexerC.LookupAssetTransactions(10).Do(context.Background())
+		case "searchForTransactions":
+			response, err = indexerC.SearchForTransactions().Do(context.Background())
+		case "any":
+			response, err = indexerC.SearchForTransactions().Do(context.Background())
+			if err == nil {
+				err = fmt.Errorf("Expected error here")
+			}
+			exptErr := fmt.Sprintf("HTTP %v", expectedStatus)
+			if strings.Contains(err.Error(), exptErr) {
+				response = nil
+				err = nil
+			} else {
+				err = fmt.Errorf("Expected error here")
+			}
+		default:
+			err = fmt.Errorf("unknown endpoint: %s", endpoint)
+		}
+	case "algod":
+		switch endpoint {
+		case "GetStatus":
+			response, err = algodC.Status().Do(context.Background())
+		case "WaitForBlock":
+			response, err = algodC.StatusAfterBlock(10).Do(context.Background())
+		case "TealCompile":
+			response, err = algodC.TealCompile([]byte{}).Do(context.Background())
+		case "RawTransaction":
+			var returnedTxid string
+			returnedTxid, err = algodC.SendRawTransaction([]byte{}).Do(context.Background())
+			response = txidresponse{TxID: returnedTxid}
+		case "GetSupply":
+			response, err = algodC.Supply().Do(context.Background())
+		case "TransactionParams":
+			var sParams types.SuggestedParams
+			sParams, err = algodC.SuggestedParams().Do(context.Background())
+			response = models.TransactionParams{
+				ConsensusVersion: sParams.ConsensusVersion,
+				Fee:              uint64(sParams.Fee),
+				GenesisID:        sParams.GenesisID,
+				Genesishash:      sParams.GenesisHash,
+				LastRound:        uint64(sParams.FirstRoundValid),
+				MinFee:           81560,
+			}
+		case "GetApplicationByID":
+			response, err = algodC.GetApplicationByID(10).Do(context.Background())
+		case "GetAssetByID":
+			response, err = algodC.GetAssetByID(10).Do(context.Background())
+		case "any":
+			response, err = indexerC.SearchForTransactions().Do(context.Background())
+			if err == nil {
+				fmt.Errorf("Expected error here")
+			}
+			if err == nil {
+				err = fmt.Errorf("Expected error here")
+			}
+			exptErr := fmt.Sprintf("HTTP %v", expectedStatus)
+			if strings.Contains(err.Error(), exptErr) {
+				response = nil
+				err = nil
+			} else {
+				err = fmt.Errorf("Expected error here")
+			}
+
+		default:
+			err = fmt.Errorf("unknown endpoint: %s", endpoint)
+		}
+	}
+	return err
+}
+
+type txidresponse struct {
+	TxID string `json:"txId"`
+}
+
+func theParsedResponseShouldEqualTheMockResponse() error {
+	var err error
+
```

Comment:
There are error message responses, is there a way to extract them and have that be part of the test verification?

Reply:
I'll return the returned error from Do. It should be nil, and returning should do.